cdc/processor/sinkmanager/worker_impl.go

```diff
@@ -83,15 +86,31 @@ func (w *workerImpl) receiveTableSinkTask(ctx context.Context, taskChan <-chan *
 			availableMem := int(requestMemSize)
 			events := make([]*model.PolymorphicEvent, 0, 1024)
 
+			lowerBound := task.lowerBound
+			upperBound := task.getUpperBound()
+
+			if w.eventCache != nil {
+				rows, size, pos := w.eventCache.pop(task.tableID)
```

Comment:
How about extracting this logic and giving it a name? Because this `for loop` gets more and more complicated.